### PR TITLE
[v2] Departments

### PR DESCRIPTION
**Pull request by regularjack**
# Added
- `departments.add`
- `departments.update`

# Changed
- Moved `departments.list` from `general.apib` to `departments.apib`

### CODE REVIEW

File: src/departments.apib [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        + data (object)
+            + id: `1` (number)
+
+### departments.update [POST /departments.update]
+
+Update a department
+
++ Request (application/json)
+
+    + Attributes (object)
```
Comment (Jdww) [inline src/departments.apib]: Q: Don't we require an id when updating and won't it be the only thing required?

File: src/datastructures.apib [hunk 2]
```diff
@@ -176,8 +176,23 @@
 + phase_id: `5` (number)
 
 ## Department (object)
-id: `1` (number)
-name: `Sales` (string)
++ name: `Human Resources` (string)
```
Comment (Jdww) [inline src/datastructures.apib]: Q: Won't we return the department id?

File: src/departments.apib (hunk 1 above)
Comment (regularjack) [inline src/departments.apib]: Correct, fixed.

File: src/datastructures.apib (hunk 2 above)
Comment (regularjack) [inline src/datastructures.apib]: Fixed

File: src/departments.apib (hunk 1 above)
Comment (Jdww) [inline src/departments.apib]: [RESOLVED by Jdww on 18/07/2017 15:28:33 :dancer:]

File: src/departments.apib [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+### departments.list [GET /departments.list]
+
+Get a list of all departments
+
++ Request (application/json)
+
+    + Attributes (object)
+        + filter (object)
+            + company_id: `1` (number) - Filter departments by a certain company.
```
Comment (stivni) [inline src/departments.apib]: Q : can we filter these by company? that doesn't make any sense to me...

File: src/departments.apib [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        + page (object)
```
Comment (stivni) [inline src/departments.apib]: Q : are we already going to build in paging? if not, you can drop it...

File: src/datastructures.apib [hunk 5]
```diff
@@ -176,8 +176,24 @@
 + phase_id: `5` (number)
 
 ## Department (object)
-id: `1` (number)
-name: `Sales` (string)
++ department_id: `42` (number)
```
Comment (stivni) [inline src/datastructures.apib]: Q : do we do `id` or `department_id` everywhere?

Comment (regularjack) [inline src/datastructures.apib]: We do `department_id` everywhere

File: src/departments.apib (hunk 3 above)
Comment (regularjack) [inline src/departments.apib]: Agree, makes no sense. Fixed.

File: src/departments.apib (hunk 4 above)
Comment (regularjack) [inline src/departments.apib]: Removed it, no paging for now.

File: apiary.apib [hunk 6]
```diff
@@ -864,8 +925,24 @@ Register a webhook.
 + phase_id: `5` (number)
 
 ## Department (object)
```
Comment (stivni) [inline apiary.apib]: B : I just made a PR to inline all datastructures that are only used once...
Let's try not to use datastructures if we don't share the structure...

File: apiary.apib [hunk 7]
```diff
@@ -864,8 +925,24 @@ Register a webhook.
...
-id: `1` (number)
-name: `Sales` (string)
++ department_id: `42` (number)
```
Comment (stivni) [inline apiary.apib]: B : I still think we should help our user by always including an `id` property...
Although I remember that we discussed using named_ids, but I don't recall what the conclusion was...

File: apiary.apib (hunk 6 above)
Comment (regularjack) [inline apiary.apib]: Inlined the datastructure. Fixed

File: apiary.apib (hunk 7 above)
Comment (regularjack) [inline apiary.apib]: Now using `id`, we seem to be using it for other data structures after all

**Comment by regularjack** [comment]
@stivni This should be good now. Can you re-review?

**Pull request merged**